File: backend/TenderGo.Models/Requests/LocationInsertRequest.cs
using System.ComponentModel.DataAnnotations;

namespace TenderGo.Models.Requests
{
    public class LocationInsertRequest
    {
        [Required]
        [StringLength(100)]
        public string Country { get; set; } = string.Empty;

        [Required]
        [StringLength(100)]
        public string Name { get; set; } = string.Empty;

        [StringLength(100)]
        public string? Region { get; set; }
    }
}

File: UI/tendergo_desktop/lib/shared/models/requests/location_insert_request.dart
class LocationInsertRequest {

  final String country;
  final String name;
  final String? region;

  const LocationInsertRequest({
    required this.country,
    required this.name,
    this.region,
  });

  Map<String, dynamic> toJson() => {
        'name': name, 
        'country': country,
        'region': region,
      };
}